metrics/statsdraw/registry.go

```diff
@@ -0,0 +1,43 @@
+package statsdraw
+
+import (
+	"fmt"
+	"time"
+
+	alstatsd "github.com/alexcesaro/statsd"
+)
+
+func NewRegistry(prefix, addr string, interval time.Duration) (*registry, error) {
+	opts := []alstatsd.Option{
+		alstatsd.Address(addr),
+		alstatsd.FlushPeriod(interval),
+	}
+	if prefix != "" {
+		opts = append(opts, alstatsd.Prefix(prefix))
+	}
+	c, err := alstatsd.New(opts...)
+	if err != nil {
+		return nil, fmt.Errorf(" cannot init statsd client: %s", err)
+	}
+	return &registry{c}, nil
+}
+
+type registry struct {
+	c *alstatsd.Client
+}
+
+func (r *registry) Names(string) []string     { return nil }
+func (r *registry) Unregister(string, string) {}
+func (r *registry) UnregisterAll(string)      {}
+
+func (r *registry) Gauge(_, name string, n float64) {
+	r.c.Count(name, n)
```

Comment:
Conflating a gauge and a counter here is going to cause problems. Counters reset between flushes unlike gauges ([at the statsd server](https://github.com/etsy/statsd/blob/master/docs/metric_types.md#gauges).)  It seems like the only time you use this is in `newRawProxy` to track the number of active conns, in order to do this correctly with a gauge you should track the actual number and report that number as a gauge.

However if you want to use the relative updates for a gauges (a questionable feature in statsd) you should always make sure that you reset the gauge value to 0 when you first update it in your app, otherwise a crash will leave you with an invalid base value.